src/test/java/org/junitpioneer/jupiter/resource/TemporaryDirectoryAssertions.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+class TemporaryDirectoryAssertions {
+
+	public static final Path ROOT_TEMP_DIR = Paths.get(System.getProperty("java.io.tmpdir"));
+
+	static void assertCanAddAndReadTextFile(Path tempDir) {
```

Comment:
This not the way we do assertions. 😭 
Instead of `assertCanAddAndReadTextFile(path)` it should look like:
```java
assertThat(path)
    .canAddTextFile()
    .canReadTextFile();
```

Reply:
But I'm confused - could this not be covered by `assertThat(path).isReadable().isWritable()`?

Reply:
Yeah, I thought that too, but I remember catching a subtle bug months ago that only unveiled by explicitly creating a file and writing to it. Admittedly I can't remember what this bug was though, so which would you prefer, `.can(Add|Read)TextFile()` or `is(Readable|Writable)()`?

(Assuming you'd prefer `.can(Add|Read)TextFile()`, I'll do some research now on how to do custom fluent assertions with AssertJ.)

Reply:
My work to translate the old assertion to `.can(Add|Read)TextFile()` has been pushed to commit [WIP - address some review feedback](https://github.com/junit-pioneer/junit-pioneer/pull/491/commits/20708c2219098b1e5646849d8e206bd0433dcb7d). Let me know what you think, such as if we should keep these assertions or stick with `.is(Readable|Writable)()`.

Reply:
I [reviewed](https://github.com/junit-pioneer/junit-pioneer/pull/491#pullrequestreview-1171790003) this part.